src/middleware/authMiddleware.js

```diff
@@ -68,23 +68,28 @@ export const authorizeRequest = (method) => {
         const targetUser = await User.findById(targetUserId);
         const targetUserRoles = targetUser ? targetUser.roles : [];
 
-        // If the target user is not the same as the current user, check permissions
+        // If the target user is not the same as the current user, check permissions on current method
         if (targetUserId && targetUserId !== req.userId) {
           hasPermission = hasPermission &&
                         rolesData.some((role) => {
                           return role.permissions.some(
                             (permission) =>
-                              targetUserRoles.every((role) =>
-                                permission.onRoles.includes(role)
-                              )
+                              permission.method === method &&
+                                    targetUserRoles.every((onRole) =>
+                                      permission.onRoles.includes(onRole)
+                                    )
```

Comment:
With this change, I think the following code is no longer necessary:

![image](https://github.com/user-attachments/assets/a96194e6-ae4a-405a-8fcd-0ff6aff03f60)

Reply:
I tested it and it is in cases where the endpoint does not include a userId as a parameter